app/scripts/directives/unit.coffee

```diff
@@ -24,10 +24,16 @@ angular.module('swarmApp').directive 'unit', ($log, game, commands, options, uti
       #util.utcdoy 1000 * secs
       if isFinite estimate.val
         nonlinear = if not (estimate.unit?.isVelocityConstant?() ? true) then 'less than ' else ''
-        secs = moment.duration(estimate.val, 'seconds').humanize()
-        if /\ years$/.test secs
-          secs = 'a long time'
-          nonlinear = ''
+        secs = moment.duration(estimate.val, 'seconds')
+        if options.showAccurateTiming()
+           secs = secs.format("Y [yr] M [mth] d [day] h [hr] m [min] s [sec]")
```

Comment:
How about a format like `[hr]:[min]:[sec]` instead of `h [hr] m [min] s [sec]`?

If something takes months/years, it's probably not useful to display hours/minutes/seconds, and it just makes the text longer (not too big a big deal, though)